File: python/my_service.py
#!/usr/bin/python3 -u
import os
import paho.mqtt.client as mqtt
import paho.mqtt.publish as publish

MQTT_HOST=os.getenv("MQTT_HOST", "test.mosquitto.org").strip()
print("Will connect to "+str(MQTT_HOST))

# The callback for when the client receives a CONNACK response from the server.
def on_connect(client, userdata, flags, rc):
    print("Connected with result code "+str(rc))

    # Subscribing in on_connect() means that if we lose the connection and
    # reconnect then subscriptions will be renewed.
    client.subscribe("uservicehack/#") # the '#' wildcard matches all topics below this one

# The callback for when a PUBLISH message is received from the server.
def on_message(client, userdata, msg):
    print(msg.topic + " " + msg.payload.decode("ascii"))

client = mqtt.Client()
client.on_connect = on_connect
client.on_message = on_message

client.connect(MQTT_HOST, 1883, 60)
client.loop_start()

# Publish a single message on the topic
publish.single("uservicehack/kittens", "minikatz are great!", hostname=MQTT_HOST)

# Blocking call that processes network traffic, dispatches callbacks and
# handles reconnecting.
# Other loop*() functions are available that give a threaded interface and a
# manual interface.
while True:
    1
client.loop_stop() # Never gets here

File: node/my_service.js
const mqtt = require('mqtt')

const MQTT_HOST = process.env.MQTT_HOST || 'test.mosquitto.org'
const service_name = 'kittens'
var client  = mqtt.connect(`mqtt://${MQTT_HOST}`)

client.on('connect', function () {
  // The '#' wildcard matches all topics below this one
  client.subscribe('uservicehack/+', function (err) {
    if (!err) {
      client.publish('uservicehack/kittens', `Service: ${service_name} has connected!`)
    }
  })
})

client.on('message', function (topic, message) {
  // message is of type Buffer, so we need to use .toString
  console.log(`Message on topic: ${topic} is: ${message.toString()}`)
  client.end() // Remove this to loop forever
})
